### ISSUE TITLE
nEdge bridge stripping vlan tag or dropping all vlan traffic

### ISSUE THREAD

**Issue by YackingCat** [ntop/ntopng#2017]
Upgraded to nEdge version 3.6.180912 - Edition.
nEdge is running in bridged mode (VLAN Trunk) between a pfSense firewall and the network.
All traffic passes through the bridge except for anything on vlan 200. Monitoring the firewall when nEdge is hooked up inline shows no traffic being received at all on the firewalls port that is configured for vlan 200. This is the exact same hardware that was used for ntopng 3.4.

---nedge.conf
auto lo
iface lo inet loopback

auto enp8s0f0
iface enp8s0f0 inet manual

auto enp8s0f1
iface enp8s0f1 inet manual

auto br0
iface br0 inet manual
        bridge_ports enp8s0f0 enp8s0f1
        bridge_stp off

auto br0:2
iface br0:2 inet static
        address 192.168.160.10
        netmask 255.255.255.0

---nedge_mgmt.conf
auto br0.200
iface br0.200 inet static
     pre-up /sbin/ip link add link br0 name br0.200 type vlan id 200
     vlan-raw-device br0
     address 192.168.6.3
     netmask 255.255.255.0

auto enp9s0f0
iface enp9s0f0 inet static
address 192.168.1.3
netmask 255.255.255.0

**Comment by simonemainardi** [comment]
can you reach the nedge web ui at 192.168.6.3? Do you see the traffic in the ui? If you sniff on enp8s0f0 and  enp8s0f1, do you see the traffic?

**Comment by YackingCat** [comment]
I cannot reach the ui at 192.168.6.3 but i can on 192.168.1.3.

As soon as i put nEdge back inline, I see both networks as local (192.168.1.0 and 192.168.6.0) in the UI. Any device that is still connected to 6.0 and still has an ip address is creating flows and every thing else but they all show data only as sent and 0 data received.
 
I can only run tcpdump at the moment on the interfaces.
Im still looking at the tcpdumps but it seems to show only outbound requests and nothing coming in to the 6.x range.

**Comment by YackingCat** [comment]
Just to avoid confusion or in case you missed it, everything not on vlan 200 is passing through the bridge without any issues.

Here is what I found after digging in the tcp dump:
Lan interface (enp8s0f0) shows anything that still has a 6.x address is still sending outbound but nothing is coming in to 6.x addresses. ARP request from 6.x to see who-has 6.1.
Wan interface (enp8s0f1) almost identical as the lan interface.
br0 shows pretty much the same thing as lan and wan interfaces
br0.200 shows ARP requests from 6.x to see who has 6.1, dhcp requests from devices that are on vlan200, and any device that still has a 6.x address is sending outbound traffic but recriving 0.

**Comment by simonemainardi** [comment]
Thank you for this detailed troubleshoot.


Can you try and leave the mgmt bridge on VLAN 200 but put it on an IP/network that is not used by the clients (just by you for the management)?

What if you set the mgmt bridge ip address on a VLAN that is different from 200?

Thank you.

**Comment by YackingCat** [comment]
I tried different VLANs, different IPs, removed VLAN200, and changing the monitoring nic. Everything I did only produced the exact same results as before.

**Comment by YackingCat** [comment]
I was kind of bored today and I decided to do a fresh install of ubuntu and nEdge just for the fun of it. Before I did that, I backed up the configuration using the backup utility in the UI. I looked inside the backup file that was created to see what was in there and I started looking at the system config file. I am seeing some strange things in there that look wrong on the way I have the current system set up or the back itself is making changes to the system.

I'm not sure is this is helpful at all or if I'm not reading it right but I thought it was worth a look.

I am using a 4 port nic.
enp8s0f0 (Lan)
enp8s0f1 (Wan)
enp9s0f0 (Unused)
enp9s0f1 (Monitoring Port)

Here is what I found that is odd:
  "interfaces":{
    "configuration":{
      "enp9s0f1":{
        "network":{
          "netmask":"255.255.255.0",
          "mode":"static",
          "ip":"192.168.1.1" <--gateway ip showing on the monitoring port
        },
        "masquerade":true,
        "family":"wired",
        "speed":{
          "upload":1000000,
          "download":1000000
        }
      },
      "enp9s0f0":{
        "network":{
          "netmask":"255.255.255.0",
          "mode":"static",
          "ip":"192.168.1.3" <--unused port showing up with the monitoring port ip
        },
        "masquerade":true,
        "family":"wired",
        "speed":{
          "upload":1000000,
          "download":1000000
        }
      },
      "enp8s0f0:1":{
        "network":{
          "netmask":"255.255.255.0",
          "mode":"static",
          "ip":"192.168.10.1"
        },
        "family":"wired",
        "speed":[]
      },
      "enp8s0f1":{
        "network":{
          "netmask":"255.255.255.0",
          "mode":"static",
          "ip":"192.168.1.1"  <--gateway showing on the wan port
        },
        "masquerade":true,
        "family":"wired",
        "speed":{
          "upload":1000000,
          "download":1000000
        }
      },
      "br0":{
        "network":{
          "netmask":"255.255.255.0",
          "gateway":"0.0.0.0",
          "mode":"vlan_trunk",
          "ip":"192.168.20.1" <--i dont even know where this comes from
        },
        "family":"bridge",
        "speed":{
          "upload":100000,
          "download":100000
        }
      },
      "enp8s0f0":{
        "network":{
          "netmask":"255.255.255.0",
          "mode":"static",
          "ip":"192.168.1.1" <--gateway showing on the lan port
        },
        "masquerade":true,
        "family":"wired",
        "speed":{
          "upload":1000000,
          "download":1000000
        }
      }
    },

**Comment by simonemainardi** [comment]
After the fresh install, are you still experiencing the same issue? Everything but VLAN 200 is working? Or did the fresh install fix the the issue?

As regards the old configuration, you don't really have to worry about it. There are some fields that are not used  / meaningful depending on the operating mode.

**Comment by YackingCat** [comment]
Everything is working as expected now. Here is what happened when I did a new install.
I am using a 4 port nic so I install ubuntu, update it to the latest kernel, and configure the 4th port to be the monitoring port. I run apt-get install nedge and it fails to start after the install. When  I look in the interfaces.d folder, there is no nedge.conf file nor did the install modify the interfaces file or create the interfaces.old file.

I changed the interface file to use the 1st port of the nic, uninstalled nedge using apt-get, and reinstalled it. After it installed, the UI was up and running.  The install created the nedge.conf file this time so i modified the monitoring port to use the 4th port on the card and rebooted. I then went into UI, set the bridge to the use the first two ports, turn on VLAN trunk, and all is good now.

The only difference between the first time I did this install is that there is no nedge_mgmt.conf file and the monitoring port is configured in the nedge.conf file.
